app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -94,9 +73,11 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
                 $immutableQuote->save();
             }
 
-            // explicitly set quote belong to guest if customer id does not exist
-            $immutableQuote
-                ->setCustomerIsGuest( (($parentQuote->getCustomerId()) ? false : true) );
+            if ($parentQuote) {
```

Comment:
can we somehow have parent quote in product page checkout? I'd like not to have this list of `if ($parentQuote)` checks